ocaml/tests/test_updateinfo.ml

```diff
@@ -1117,11 +1132,295 @@ module UpdateInfoOfXml = Generic.MakeStateless (struct
                   ; description= "description"
                   ; rec_guidance= None
                   ; abs_guidance= None
+                  ; guidance= [(Livepatch, [RestartToolstack])]
+                  ; guidance_applicabilities= []
+                  ; spec_info= "special information"
+                  ; url= "https://update.details.info"
+                  ; update_type= "security"
+                  ; livepatch_guidance= Some RestartToolstack
+                  ; livepatches= []
+                  ; issued= Xapi_stdext_date.Date.epoch
+                  ; severity= Severity.None
+                  }
+              )
+            ]
+        )
+      ; (* guidance in new format: empty guidance *)
+        ( {|
+            <updates>
+              <update type="security">
+                <id>UPDATE-0000</id>
+                <title>title</title>
+                <summary>summary</summary>
+                <description>empty guidance</description>
+                <special_info>special information</special_info>
+                <url>https://update.details.info</url>
+                <recommended_guidance>EvacuateHost</recommended_guidance>
+                <absolute_guidance>RebootHost</absolute_guidance>
+                <guidance_applicabilities/>
+                <guidance/>
+                <livepatch_guidance>RestartDeviceModel</livepatch_guidance>
+                <livepatches>
+                  <livepatch component="xen" base="4.19.19-8.0.20.xs8" to="4.19.19-8.0.21.xs8" base-buildid=""/>
+                  <livepatch component="kernel" base="4.19.19-8.0.20.xs8" base-buildid="9346194f2e98a228f5a595b13ecabd43a99fada0"/>
+                </livepatches>
+              </update>
+            </updates>
+          |}
+        , Ok
+            [
+              ( "UPDATE-0000"
+              , UpdateInfo.
+                  {
+                    id= "UPDATE-0000"
+                  ; summary= "summary"
+                  ; description= "empty guidance"
+                  ; rec_guidance= Some EvacuateHost
+                  ; abs_guidance= Some RebootHost
+                  ; guidance=
+                      [
+                        (Mandatory, [EvacuateHost])
+                      ; (Full, [RebootHost])
+                      ; (Livepatch, [RestartDeviceModel])
+                      ]
+                  ; guidance_applicabilities= []
+                  ; spec_info= "special information"
+                  ; url= "https://update.details.info"
+                  ; update_type= "security"
+                  ; livepatch_guidance= Some RestartDeviceModel
+                  ; livepatches= []
+                  ; issued= Xapi_stdext_date.Date.epoch
+                  ; severity= Severity.None
+                  }
+              )
+            ]
+        )
+      ; (* guidance in new format: empty mandatory and full *)
```

Comment:
How about a test case with only new `guidance`, but without old `recommended_guidance`, `absolute_guidance`?

Reply:
Sorry, I can't see a new test case added. We'd better add it now as there would have no xapi code change when the old updates fields are removed by factory.